### ISSUE TITLE
[Bug]:  Error after upgrading my NVIM  to 0.10

### ISSUE THREAD

**Issue by Prashant20nov2003** [akinsho/bufferline.nvim#906]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

After upgrading my nvim to latest version getting bunch of errors.

### What did you expect to happen?

E5108: Error executing lua ...al/share/nvim/lazy/bufferline.nvim/lua/bufferline/ui.lua:119: Segments must be a list
stack traceback:
        [C]: in function 'assert'
        ...al/share/nvim/lazy/bufferline.nvim/lua/bufferline/ui.lua:119: in function 'get_component_size'
        ...al/share/nvim/lazy/bufferline.nvim/lua/bufferline/ui.lua:480: in function <...al/share/nvim/lazy/bufferline.nvim/lua/bufferline/ui.lua:
461>
        vim/shared.lua: in function 'get_components'
        ...local/share/nvim/lazy/bufferline.nvim/lua/bufferline.lua:56: in function <...local/share/nvim/lazy/bufferline.nvim/lua/bufferline.lua:5
4>

### Config

return {
	"akinsho/bufferline.nvim",
	dependencies = { "nvim-tree/nvim-web-devicons" },
	version = "*",
	opts = {
		options = {
			mode = "tabs",
		},
	},
}

### Additional Information

...

**Comment by sysradium** [comment]
I guess this is related:
https://github.com/akinsho/bufferline.nvim/commit/a6ad228f77c276a4324924a6899cbfad70541547

Current version on brew has that issue:
```
➜ nvim --version
NVIM v0.10.0
Build type: Release
LuaJIT 2.1.1713773202
```

**Comment by Prashant20nov2003** [comment]
Alex 
I use Ubuntu 
I have seen people on reddit getting the same error
even on the main branch.

**Comment by dmehala** [comment]
I experienced the same issue, using the main branch fixed it. Thanks.

**Comment by sysradium** [comment]
@Prashant20nov2003 

I would assume that `vim.fn.has("nvim-0.10"))` check wrongly fails for you.
You can check neovim outputs for those commands:
```
// should not work
:lua print(vim.tbl_isarray)
:lua print(vim.isarray)
:lua print(vim.version())
```

**Comment by akinsho** [comment]
Thanks all for flagging this issue,
I'm closing this and other v0.9.0 -> 0.10.0 migration issues as this has now been fixed on the latest release `4.6.0`

**Comment by Prashant20nov2003** [comment]
Thanks for fixing this issue.
